File: resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/TestConstants.scala
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.apache.spark.deploy.k8s.integrationtest

object TestConstants {
  val BACKEND_MINIKUBE = "minikube"
  val BACKEND_DOCKER_FOR_DESKTOP = "docker-for-desktop"
  val BACKEND_DOCKER_DESKTOP = "docker-desktop"
  val BACKEND_CLOUD = "cloud"

  val CONFIG_KEY_DEPLOY_MODE = "spark.kubernetes.test.deployMode"
  val CONFIG_KEY_KUBE_CONFIG_CONTEXT = "spark.kubernetes.test.kubeConfigContext"
  val CONFIG_KEY_KUBE_MASTER_URL = "spark.kubernetes.test.master"
  val CONFIG_KEY_KUBE_NAMESPACE = "spark.kubernetes.test.namespace"
  val CONFIG_KEY_KUBE_SVC_ACCOUNT = "spark.kubernetes.test.serviceAccountName"
  val CONFIG_KEY_IMAGE_JVM = "spark.kubernetes.test.jvmImage"
  val CONFIG_KEY_IMAGE_PYTHON = "spark.kubernetes.test.pythonImage"
  val CONFIG_KEY_IMAGE_R = "spark.kubernetes.test.rImage"
  val CONFIG_KEY_IMAGE_TAG = "spark.kubernetes.test.imageTag"
  val CONFIG_KEY_IMAGE_TAG_FILE = "spark.kubernetes.test.imageTagFile"
  val CONFIG_KEY_IMAGE_REPO = "spark.kubernetes.test.imageRepo"
  val CONFIG_KEY_UNPACK_DIR = "spark.kubernetes.test.unpackSparkDir"
}

File: remote-shuffle-service/src/test/java/org/apache/spark/remoteshuffle/testutil/TestConstants.java
/*
 * This file is copied from Uber Remote Shuffle Service
 * (https://github.com/uber/RemoteShuffleService) and modified.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *     http://www.apache.org/licenses/LICENSE-2.0
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.apache.spark.remoteshuffle.testutil;

import org.apache.spark.remoteshuffle.clients.ShuffleWriteConfig;

public class TestConstants {
  public static final int NETWORK_TIMEOUT = 30000;

  public static final int DATA_AVAILABLE_POLL_INTERVAL = 10;
  public static final int DATA_AVAILABLE_TIMEOUT = 30000;

  public static final int COMPRESSION_BUFFER_SIZE = 64 * 1024;

  public static final ShuffleWriteConfig SHUFFLE_WRITE_CONFIG = new ShuffleWriteConfig((short) 3);

  public static final long CONNECTION_IDLE_TIMEOUT_MILLIS = 30 * 1000;
}
